base/number.jl

```diff
@@ -3,6 +3,15 @@
 ## generic operations on numbers ##
 
 isinteger(x::Integer) = true
+iszero(x::Number) = x == 0 # faster than x == zero(x) for complex, bigfloat, etc.
+
+"""
+    iszero(x)
+
+Return `true` if `x == zero(x)`; if `x` is an array, this checks whether
+all of the elements of `x` are zero.
+"""
+iszero(x) = x == zero(x) # fallback method
```

Comment:
Why wouldn't e.g. `iszero("cat") == false` be sensible? `"cat"` certainly isn't a `zero`, and for e.g. #19589 that behavior is necessary.